src/main/java/org/springframework/data/couchbase/core/convert/AbstractCouchbaseConverter.java

```diff
@@ -160,15 +160,16 @@ public Object convertForWriteIfNeeded(Object inValue) {
 		Class<?> elementType = value.getClass();
 
 		if (elementType == null || conversions.isSimpleType(elementType)) {
-			// superseded by EnumCvtrs value = Enum.class.isAssignableFrom(value.getClass()) ? ((Enum<?>) value).name() : value;
+			// superseded by EnumCvtrs value = Enum.class.isAssignableFrom(value.getClass()) ? ((Enum<?>) value).name() :
+			// value;
 		} else if (value instanceof Collection || elementType.isArray()) {
 			TypeInformation<?> type = ClassTypeInformation.from(value.getClass());
 			value = ((MappingCouchbaseConverter) this).writeCollectionInternal(MappingCouchbaseConverter.asCollection(value),
 					new CouchbaseList(conversions.getSimpleTypeHolder()), type, null, null);
 		} else {
 			CouchbaseDocument embeddedDoc = new CouchbaseDocument();
 			TypeInformation<?> type = ClassTypeInformation.from(value.getClass());
-			((MappingCouchbaseConverter) this).writeInternalRoot(value, embeddedDoc, type, false, null);
+			((MappingCouchbaseConverter) this).writeInternalRoot(value, embeddedDoc, type, false, null, true);
 			value = embeddedDoc;
```

Comment:
writeInternalRoot needs a flag of whether or not to process PropertyValueConverters (specifically the CryptoConverter) so it doesn't recursively encrypt such an annotated property.